lib/matplotlib/quiver.py

```diff
@@ -626,19 +626,19 @@ def _angles_lengths(self, U, V, eps=1):
 
     def _make_verts(self, U, V, angles):
         uv = (U + V * 1j)
-        str_angles = isinstance(angles, six.string_types)
-        if str_angles and (angles == 'xy' and self.scale_units == 'xy'):
+        str_angles = angles if isinstance(angles, six.string_types) else ''
+        if str_angles == 'xy' and self.scale_units == 'xy':
             # Here eps is 1 so that if we get U, V by diffing
             # the X, Y arrays, the vectors will connect the
             # points, regardless of the axis scaling (including log).
             angles, lengths = self._angles_lengths(U, V, eps=1)
-        elif str_angles and (angles == 'xy' or self.scale_units == 'xy'):
+        elif str_angles == 'xy' or self.scale_units == 'xy':
             # Calculate eps based on the extents of the plot
             # so that we don't end up with roundoff error from
             # adding a small number to a large.
             eps = np.abs(self.ax.dataLim.extents).max() * 0.001
             angles, lengths = self._angles_lengths(U, V, eps=eps)
-        if self.scale_units == 'xy':
+        if str_angles and self.scale_units == 'xy':
```

Comment:
well, all the scalings except 'xy' are based on U and V, so I guess I it sufficient to leave it like that. throwing exception on `str_angles not in ('xy', 'uv')` seems to be an overkill since `a` and `theta` will fallback to using `U` and `V` values ...